IM_calculation/Advanced_IM/scripts/overview_adv_IM_status.py

```diff
@@ -0,0 +1,135 @@
+}
+
+
+def process(type, model, sim_root, outdir, pattern=None, verbose=False):
+    search_pattern = PATTERN_TEMPLATE.get(type, pattern) + f"/{model}_status.csv"
+    print(f"Searching for {sim_root}/{search_pattern}")
+
+    event_status_files = list(sim_root.glob(search_pattern))
+
+    if len(event_status_files) == 0:
+        print(f"Error:No status files found at {sim_root}/{search_pattern}")
+        print(f"Check the sim_root, model and search pattern")
+        sys.exit(0)
+
+    event_status_dict = {}
+    for event_status_file in event_status_files:
+        if type == "sim":
+            event_name = Path(event_status_file).parents[2].name
+        else:
+            event_name = Path(event_status_file).parent.name
+        event_status_df = pd.read_csv(event_status_file)
+        event_status_dict[event_name] = {"total": 0}
+        for status in ALL_STATUS:
+
+            count = event_status_df["status"].value_counts().get(status, 0)
+            event_status_dict[event_name][status] = count
+            event_status_dict[event_name]["total"] += count
+
+    overall_status_df = pd.DataFrame.from_dict(
+        event_status_dict, orient="index", columns=["total"] + ALL_STATUS
+    )
+
+    overall_status_df.to_csv(outdir / f"overall_status_{model}_{type}.csv")
+
+    not_finished_df = overall_status_df.loc[
+        overall_status_df["total"] != overall_status_df["finished"]
+    ]
+    print("------------")
+    print(
+        overall_status_df.sum(axis=0).to_string(header=None)
+    )  # .to_string(header=None) removes dtype: int64
+    print("------------")
+    if len(not_finished_df) > 0:
+        print(
+            f"{len(not_finished_df)} incomplete/unsuccessful event(s) found (Total: {len(overall_status_df)})"
+        )
+        if verbose:
+            print(not_finished_df)
+    else:
+        print(f"All completed successfully! (Total: {len(overall_status_df)})")
+    return overall_status_df
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "sim_root",
+        type=Path,
+        help="Root path of GM sim.ie. Parent directory of Runs or ObservedGroundMotions",
+    )
+    parser.add_argument("model", help="Name of the advanced IM model")
+    parser.add_argument(
+        "type",
+        choices=TYPES,
+        default="custom",
+        help="Type of data. If neither sim or obs, choose custom and provide --pattern",
+    )
+    parser.add_argument(
+        "--outdir", type=Path, default=Path.cwd(), help="Output directory"
+    )
+    parser.add_argument(
+        "--pattern",
+        default=None,
+        help='Status file search pattern for custom type. eg.) "Runs/*/*/IM_calc"',
+    )
+
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="store_true",
+        default=False,
+        help="Print unfinished events summary",
+    )
+
+    args = parser.parse_args()
+
+    if not args.sim_root.is_dir():
```

Comment:
Would a direct assert statement be better instead of raising an assert if [conditional]?

Reply:
Perhaps less helpful to user? 
```
Traceback (most recent call last):
  File "overview_adv_IM_status.py", line 137, in <module>
    main()
  File "overview_adv_IM_status.py", line 116, in main
    assert args.sim_root.is_dir()
AssertionError

```

Reply:
`assert args.sim_root.is_dir(), "Provide a valid directory path"` ?

Just don't put brackets around it because tuples are True. I've made that mistake before.

Reply:
Done